### PR TITLE
Surfacing various connection states

### PR DESCRIPTION
**Pull request by kiiadi**
Improving visibility to current AWS connection state, surfacing it with consistent messaging.
Untitled

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/AwsSettingsPanel.kt [hunk 1]
```diff
@@ -57,18 +48,17 @@ private class AwsSettingsPanel(private val project: Project) : StatusBarWidget,
 
     override fun install(statusBar: StatusBar) {
         this.statusBar = statusBar
-        project.messageBus.connect().subscribe(ProjectAccountSettingsManager.CONNECTION_SETTINGS_CHANGED, this)
+        project.messageBus.connect().subscribe(ProjectAccountSettingsManager.CONNECTION_SETTINGS_STATE_CHANGED, this)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/AwsSettingsPanel.kt:51]: This should actually be `connect(this)`

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/AwsSettingsPanel.kt]: fixed

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties [hunk 2]
```diff
@@ -621,8 +621,14 @@ settings.credentials=Credentials
 settings.credentials.none_selected=No credentials selected
 settings.credentials.profile_sub_menu=All Credentials
 settings.credentials.recent=Recent Credentials
+settings.none_selected=No region or credentials selected
 settings.partitions=Partitions
 settings.regions.none_selected=No region selected
 settings.regions.recent=Recent Regions
 settings.regions.region_sub_menu=All Regions
+settings.states.initializing=Toolkit initializing...
+settings.states.invalid=Unable to connect to AWS using selected credentials ({0}) / region ({1}): {2}
```
Comment (abrooksv) [inline resources/resources/software/aws/toolkits/resources/localized_messages.properties:630]: Can we put a new line after the colon

**Comment by abrooksv** [comment]
So here is a question:

Do we consider the validating / error states to be considered empty states of the explorer?

If so, should we follow their guidance? https://jetbrains.github.io/ui/principles/empty_state/

The state machine changes I like, and have no concerns with.

File: resources/resources/software/aws/toolkits/resources/localized_messages.properties (hunk 2 above)
Comment (kiiadi) [reply resources/resources/software/aws/toolkits/resources/localized_messages.properties]: done

**Pull request merged**